com.unity.render-pipelines.high-definition/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl

```diff
@@ -1,484 +1,243 @@
-        // NPR work.
-        //
-        // The small offset at the end reduces the "going to zero" impact
-        // where this is really close to exactly opposite
-        weights[i] = Sq(dot(-probeToSampleUnbiasedDirectionWS, sampleNormalWS) * 0.5 + 0.5) + 0.2;
-
-        float2 probeOctahedralDepthUV = PackNormalOctRectEncode(probeToSampleBiasedDirectionWS);
-        int2 probeVolumeTexel2DMin = (probeVolumeTexel3DOffset.z == 0.0) ? probeVolumeTexel2DMinBack : probeVolumeTexel2DMinFront;
-        float2 probeOctahedralDepthTexel2D = (probeOctahedralDepthUV + probeVolumeTexel3DOffset.xy) * (float)OCTAHEDRAL_DEPTH_RESOLUTION + (float2)probeVolumeTexel2DMin;
-        float2 probeOctahedralDepthAtlasUV = probeOctahedralDepthTexel2D * probeVolumeAtlasOctahedralDepthResolutionAndInverse.zw + probeVolumeOctahedralDepthScaleBias.zw;
-
-        float2 temp = SAMPLE_TEXTURE2D_LOD(_ProbeVolumeAtlasOctahedralDepth, s_linear_clamp_sampler, probeOctahedralDepthAtlasUV, 0).xy;
-        float mean = temp.x;
-        float variance = temp.y;
-
-        // http://www.punkuser.net/vsm/vsm_paper.pdf; equation 5
-        // Need the max in the denominator because biasing can cause a negative displacement
-        float chebyshevWeight = variance / (variance + Sq(max(probeToSampleBiasedDistanceWS - mean, 0.0)));
-
-        // Increase contrast in the weight
-        chebyshevWeight = max(chebyshevWeight * chebyshevWeight * chebyshevWeight, 0.0);
-
-        // Avoid visibility weights ever going all of the way to zero because when *no* probe has
-        // visibility we need some fallback value.
-        weights[i] *= max(0.2, ((probeToSampleBiasedDistanceWS <= mean) ? 1.0 : chebyshevWeight));
-
-        // Avoid zero weight
-        weights[i] = max(0.000001, weights[i]);
-
-        // A tiny bit of light is really visible due to log perception, so
-        // crush tiny weights but keep the curve continuous.
-        const float CRUSH_THRESHOLD = 0.2;
-        if (weights[i] < CRUSH_THRESHOLD)
-        {
-            weights[i] *= weights[i] * weights[i] * (1.0 / Sq(CRUSH_THRESHOLD));
-        }
-
-        // Aggressively prevent weights from going anywhere near 0.0f no matter
-        // what the compiler (or, for that matter, the algorithm) thinks.
-        const bool RECURSIVE = true;
-        weights[i] = clamp(weights[i], (RECURSIVE ? 0.1 : 0.0), 1.01);
-    }
+    return (l1 - 0.5) * 4.0 * l0;
 }
-#endif
 
-void ProbeVolumeComputeOBBBoundsToFrame(OrientedBBox probeVolumeBounds, out float3x3 obbFrame, out float3 obbExtents, out float3 obbCenter)
+float3 EvaluateAmbientProbe(float3 normalWS)
 {
-    obbFrame = float3x3(probeVolumeBounds.right, probeVolumeBounds.up, cross(probeVolumeBounds.right, probeVolumeBounds.up));
-    obbExtents = float3(probeVolumeBounds.extentX, probeVolumeBounds.extentY, probeVolumeBounds.extentZ);
-    obbCenter = probeVolumeBounds.center;
+    real4 SHCoefficients[7];
+    SHCoefficients[0] = unity_SHAr;
```

Comment:
We should create a function with this one, the exact same code is used in builtinGIUtilities.hlsl (next PR)

Reply:
This the function :) 
I just need to move this to the builtinGIUtilities and use there.